### PR TITLE
skyapm dotnet support log collecting

### PR DESCRIPTION
**Pull request by GeorGeWzw**
Please answer these questions before submitting pull request
___
### skywalking link tracing log
- Dotnet program connects to skywalking link tracing log.

### CODE REVIEW

File: src/SkyApm.Agent.AspNetCore/SkyApmHostingStartup.cs [hunk 1]
```diff
@@ -29,7 +29,11 @@ internal class SkyApmHostingStartup : IHostingStartup
     {
         public void Configure(IWebHostBuilder builder)
         {
-            builder.ConfigureServices(services => services.AddSkyAPM(ext => ext.AddAspNetCoreHosting()));
+            builder.ConfigureServices(services => services.AddSkyAPM(ext => ext.AddAspNetCoreHosting())
+            .AddLogging(logging =>
+            {
+                logging.AddSkyApmLogger();
+            }));
```
Comment (liuhaoyang) [inline src/SkyApm.Agent.AspNetCore/SkyApmHostingStartup.cs:36]: Is it necessary to report logs to Skywalking as the default option?

File: src/SkyApm.Core/SkyLogging/SkyApmLogger.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
+﻿using Microsoft.Extensions.Logging;
```
Comment (liuhaoyang) [inline src/SkyApm.Core/SkyLogging/SkyApmLogger.cs:1]: The `SkyApm.Core` project should not depend on `Microsoft.Extensions.Logging`

Comment (liuhaoyang) [reply src/SkyApm.Core/SkyLogging/SkyApmLogger.cs:1]: A better approach is to implement the Skywalking-log adapter in a new solution project

**Review by liuhaoyang — CHANGES_REQUESTED** [review]

File: sample/Sky.Apm.Sample.Logging/Controllers/WeatherForecastController.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Mvc;
```
Comment (liuhaoyang) [inline sample/Sky.Apm.Sample.Logging/Controllers/WeatherForecastController.cs:1]: Please add copyright header

File: sample/Sky.Apm.Sample.Logging/Program.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+//using Sky.Apm.Sample.Logging.Test;
```
Comment (liuhaoyang) [inline sample/Sky.Apm.Sample.Logging/Program.cs:1]: Please add copyright header

**Comment by liuhaoyang** [comment]
Please add a copyright header to all new files

**Comment by wu-sheng** [comment]
> Please add a copyright header to all new files

@liuhaoyang Please consider skywalking-eyes to this repository.

File: src/SkyApm.Abstractions/Logging/ILoggerFactory.cs [hunk 5]
```diff
@@ -23,5 +23,6 @@ namespace SkyApm.Logging
     public interface ILoggerFactory
     {
         ILogger CreateLogger(Type type);
+        SkyApm.Logging.ILogger CreateSkyApmLogger(Type type);
```
Comment (ElderJames) [inline src/SkyApm.Abstractions/Logging/ILoggerFactory.cs:26]:
```suggestion
        ILogger CreateSkyApmLogger(Type type);
```

File: src/SkyApm.Core/Logging/NullLoggerFactory.cs [hunk 6]
```diff
@@ -26,5 +26,10 @@ public ILogger CreateLogger(Type type)
         {
             return new NullLogger();
         }
+
+        public ILogger CreateSkyApmLogger(Type type)
+        {
+            throw new NotImplementedException();
```
Comment (ElderJames) [inline src/SkyApm.Core/Logging/NullLoggerFactory.cs:32]: Why not implement the method?

File: src/SkyApm.Core/SkyApm.Core.csproj [hunk 7]
```diff
@@ -12,6 +12,11 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <DefineConstants Condition="'$(TargetFramework)' == 'netcoreapp2.1'">$(DefineConstants);SPAN</DefineConstants>
   </PropertyGroup>
+  <!--<ItemGroup>
+    <Compile Remove="SkyLogging\**" />
+    <EmbeddedResource Remove="SkyLogging\**" />
+    <None Remove="SkyLogging\**" />
+  </ItemGroup>-->
```
Comment (ElderJames) [inline src/SkyApm.Core/SkyApm.Core.csproj:19]: Please remove the comment code.

File: src/SkyApm.Utilities.Logging/SkyApmLogger.cs [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        private readonly Type _loggerName;
+        private readonly IEntrySegmentContextAccessor _entrySegmentContextAccessor;
+        private readonly ISkyApmLogDispatcher _skyApmLogDispatcher;
+        private readonly bool _pushSkywalking;
+        public SkyApmLogger(Type type, IServiceProvider serviceProvider,bool pushSkywalking)
+        {
+            _entrySegmentContextAccessor = serviceProvider.GetService<IEntrySegmentContextAccessor>();
+            _skyApmLogDispatcher = serviceProvider.GetService<ISkyApmLogDispatcher>();
+            _loggerName = type;
+            _pushSkywalking= pushSkywalking;
```
Comment (ElderJames) [inline src/SkyApm.Utilities.Logging/SkyApmLogger.cs:40]: I don't understand what the meaning of this switch is.

Comment (GeorGeWzw) [reply src/SkyApm.Utilities.Logging/SkyApmLogger.cs:40]: Users can decide whether to push logs to Skywallking

**Review by liuhaoyang — APPROVED** [review]

**Pull request merged**